### PR TITLE
apis: define PodLogWatch data model

### PR DESCRIPTION
**Pull request by nicks**
Hello @milas,

Please review the following commits I made in branch nicks/ch11684:

610ce0eade9680d17928c02b417155f719756d21 (2021-03-19 17:04:14 -0400)
apis: define PodLogWatch data model

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
most of this PR is autogenerated. the main bit is the data models

File: pkg/apis/core/v1alpha1/podlogstream_types.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+	Items []PodLogStream `json:"items"`
+}
+
+// PodLogStreamSpec defines the desired state of PodLogStream
+//
+// Translated into a PodLog query to the current Kubernetes cluster:
+// https://pkg.go.dev/k8s.io/api/core/v1#PodLogOptions
+//
+// TODO(nick): Should all Kubernetes types have an object that describes
```
Comment (milas) [inline pkg/apis/core/v1alpha1/podlogstream_types.go:62]: I wonder if we should have a helper type that's a tuple of `Kind, Namespace, Name` - kind of a `types.NamespacedName++`

File: pkg/apis/core/v1alpha1/podlogstream_types.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+// which Kubernetes context to use?
+type PodLogStreamSpec struct {
+	// The name of the pod to watch. Required.
+	Pod string `json:"pod,omitempty"`
```
Comment (milas) [inline pkg/apis/core/v1alpha1/podlogstream_types.go:66]: I _think_ including `omitempty` might make it implicitly not required if omitted; I was sorta seeing behavior like that before. But I didn't really mess with `+optional` etc that much

File: pkg/apis/core/v1alpha1/podlogstream_types.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+	// The namespace of the pod to watch. Defaults to the kubecontext default namespace.
+	//
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// An RFC3339 timestamp from which to show logs. If this value
+	// precedes the time a pod was started, only logs since the pod start will be returned.
+	// If this value is in the future, no logs will be returned.
+	// Only one of sinceSeconds or sinceTime may be specified.
```
Comment (milas) [inline pkg/apis/core/v1alpha1/podlogstream_types.go:76]: Outdated (re: `sinceSeconds`)?

**Review by milas — APPROVED** [review]

File: pkg/apis/core/v1alpha1/podlogstream_types.go (hunk 1 above)
Comment (nicks) [reply pkg/apis/core/v1alpha1/podlogstream_types.go:62]: oh i think this is weirder tho - this is like a KUBECONFIG, context, and namespace

File: pkg/apis/core/v1alpha1/podlogstream_types.go (hunk 2 above)
Comment (nicks) [reply pkg/apis/core/v1alpha1/podlogstream_types.go:66]: guess we'll find out! :man_shrugging:

File: pkg/apis/core/v1alpha1/podlogstream_types.go (hunk 3 above)
Comment (nicks) [reply pkg/apis/core/v1alpha1/podlogstream_types.go]: oops, ya, copy-paste error

**Pull request merged**